Lib/_weakrefset.py

```diff
@@ -90,7 +90,13 @@ def clear(self):
         self.data.clear()
 
     def copy(self):
-        return self.__class__(self)
+        new = WeakSet()
```

Comment:
`self.__class__(self)` doesn't call _commit_removals like the Weak Dictionary versions do

```suggestion
        if self._pending_removals:
            self._commit_removals()
        new = WeakSet()
```

Reply:
Yes, that would be good becauseL:
1. there's potentially less state to copy that way;
2. since you're not using `_IterationGuard` anymore, the previous indirect call to `_commit_removals()` was removed.